src/ui/CategoryFacet/CategoryValue.ts

```diff
@@ -0,0 +1,85 @@
+import { Dom, $$ } from '../../utils/Dom';
+import { CategoryFacetTemplates } from './CategoryFacetTemplates';
+import { CategoryChildrenValueRenderer } from './CategoryValueChildrenRenderer';
+import { CategoryFacet } from './CategoryFacet';
+import { ICategoryFacetValue } from '../../rest/CategoryFacetValue';
+import { analyticsActionCauseList } from '../Analytics/AnalyticsActionListMeta';
+
+export interface CategoryValueParent {
+  path: string[];
+  categoryChildrenValueRenderer: CategoryChildrenValueRenderer;
+  renderAsParent(categoryFacetValue: ICategoryFacetValue): CategoryValue;
+  renderChildren(categoryFacetValues: ICategoryFacetValue[]): void;
+}
+
+export class CategoryValue implements CategoryValueParent {
+  private collapseArrow: Dom;
+  private labelOnClick: (e: Event) => void;
+  private label: Dom;
+
+  public element: Dom;
+  public isActive = false;
+  public categoryChildrenValueRenderer: CategoryChildrenValueRenderer;
+
+  constructor(
+    private parentElement: Dom,
+    public value: string,
+    public count: number,
+    private categoryFacetTemplates: CategoryFacetTemplates,
+    private categoryFacet: CategoryFacet,
+    public path: string[]
+  ) {
+    this.element = this.categoryFacetTemplates.buildListElement({ value: this.value, count: this.count });
+    this.label = $$(this.element.find('.coveo-category-facet-value-label'));
+    this.collapseArrow = this.categoryFacetTemplates.buildCollapseArrow();
+    this.categoryChildrenValueRenderer = new CategoryChildrenValueRenderer(this.element, categoryFacetTemplates, this, this.categoryFacet);
+    this.labelOnClick = () => {
+      if (!this.passededMaximumDepth()) {
+        this.onLabelClick();
+      }
+    };
+    this.label.one('click', this.labelOnClick);
```

Comment:
You are not binding `this`, using fat arrow or `bind(this)`..

Reply:
this.labelOnClick was already a function defined with a fat arrow